code-challenges.test.js

```diff
@@ -24,31 +36,112 @@ var colors2 = ["chartreuse", "indigo", "periwinkle", "ochre", "aquamarine", "saf
+
+
 
 
 // --------------------2) Create a function that takes an array of numbers and returns an array of the minimum and maximum numbers in that order.
 
 // a) Create a test with expect statements for each of the variables provided.
 
-var nums1 = [3, 56, 90, -8, 0, 23, 6]
-// Expected output: [-8, 90]
-var nums2 = [109, 5, 9, 67 8, 24]
-// Expected output: [5, 109]
+//describe method that lists the name of the function
+//test method describes what function does
+//an expect method calls the function to test arguments
 
+describe("minMax",() => {
+    it('takes an array of numbers and returns an array of the minimum and maximum numbers in that order',() =>{
+    expect(minMax(nums1)).toEqual([-8, 90])
+    expect(minMax(nums2)).toEqual([5, 109])
+    })
+})
+
+
+ var nums1 = [3, 56, 90, -8, 0, 23, 6]
+ // Expected output: [-8, 90]
+ var nums2 = [109, 5, 9, 67, 8, 24]
+ // Expected output: [5, 109]
 
 
-// b) Create the function that makes the test pass.
+
+// // b) Create the function that makes the test pass.
+
+//declare a function named minMax
+//declare variables maxValues and minValues to array[0]
+//for loop for iterating the array length
+//if and else if to check the max and min values in array
+
+const minMax = (array) =>{
+    let maxValues=array[0]
+    let minValues=array[0]
+ 
+    for (let i = 0; i < array.length; i++) {
+      if(array[i] < minValues){
+        minValues = array[i];
+      }
+      else if(array[i] >maxValues){
+        maxValues = array[i];
+      }
+      }
+    return [minValues,maxValues]
+    
+  }
+
+  console.log(minMax(nums1));
+  console.log(minMax(nums2));
 
 
 
 // --------------------3) Create a function that takes in two arrays as arguments and returns one array with no duplicate values. STRETCH: Use the spread operator to pass in a dynamic number of arguments.
 
 // a) Create a test with an expect statement using the variables provided.
 
-var testArray1 = [3, 7, 10, 5, 4, 3, 3]
-var testArray2 = [7, 8, 2, 3, 1, 5, 4]
-// Expected output: [3, 7, 10, 5, 4, 8, 2, 1]
+//describe method that lists the name of the function
+//test method describes what function does
+//an expect method calls the function to test arguments
 
+describe("duplicateValue",() => {
+    it('function that takes in two arrays as arguments and returns one array with no duplicate',() => {
+        expect(duplicateValue(testArray1,testArray2)).toEqual([3, 7, 10, 5, 4, 8, 2, 1])
+        })
+})
 
 
-// b) Create the function that makes the test pass.
+var testArray1 = [3, 7, 10, 5, 4, 3, 3]
+var testArray2 = [7, 8, 2, 3, 1, 5, 4]
+ // Expected output: [3, 7, 10, 5, 4, 8, 2, 1]
+
+
+
+ // b) Create the function that makes the test pass.
+
+ //declare a function named duplicateValue
+ //declare variable named testArray3 
+ //concat is used to combine the testArray1 and testArray2
+ //for loop to iterate 
+//.splice is used to change the contents of an array by romoving or replacing existing elements
```

Comment:
I saw a similar work up on stack overflow! This is a good start to a write up of what is going on here. I'd like to see more details in the pseudo code but the solution is very nice!